test/js_test.go

```diff
@@ -3709,19 +3710,19 @@ func TestJetStream_Unsubscribe(t *testing.T) {
 		deleteAllConsumers(t)
 	})
 
-	t.Run("durable consumers not deleted on drain", func(t *testing.T) {
-		subB, err := js.SubscribeSync("foo.B", nats.Durable("B"))
+	t.Run("durable consumers deleted on drain", func(t *testing.T) {
+		subB, err := js.Subscribe("foo.B", func(_ *nats.Msg) {}, nats.Durable("B"), nats.DeleteConsumer())
```

Comment:
All these changes till needed?

Reply:
Meaning? These tests are about checking that consumers are deleted, but the semantic have changed a bit. I could refactor them completely to match the current expectations as opposed to try "fixing" them. Will do that tomorrow.

Reply:
Sounds good. Looking at the examples, would you add the opt to Unsubscribe() vs Subscribe()?

Reply:
Although it would make more sense, I would not change Unsubscribe()/Drain() (even adding var option may cause backward compatibility, and may not be possible in other lang).